grid-proxy/internal/indexer/indexer.go

```diff
@@ -0,0 +1,54 @@
+package indexer
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/tfgrid-sdk-go/rmb-sdk-go/peer"
+)
+
+const (
+	indexerCallTimeout = 10 * time.Second
+)
+
+type Watcher interface {
+	Start(ctx context.Context)
```

Comment:
Should be Starter and Stopper and Pauser interfaces instead of Indexer or Watcher here